service/worker/sysworkflow/initiator.go

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package sysworkflow
+
+import (
+	"context"
+	"fmt"
+	"github.com/uber/cadence/client/frontend"
+	"github.com/uber/cadence/common/service/dynamicconfig"
+	"go.uber.org/cadence/client"
+	"math/rand"
+	"time"
+)
+
+type (
+
+	// Initiator is used to trigger system tasks
+	Initiator interface {
+		Archive(request *ArchiveRequest) error
+	}
+
+	initiator struct {
+		cadenceClient client.Client
+		numSWFn       dynamicconfig.IntPropertyFn
+	}
+
+	// Signal is the data sent to system tasks
+	Signal struct {
+		RequestType    RequestType
+		ArchiveRequest *ArchiveRequest
+	}
+
+	// ArchiveRequest signal used for archival task
+	ArchiveRequest struct {
+		UserWorkflowID string
+		UserRunID      string
+	}
+)
+
+// NewInitiator creates a new Initiator
+func NewInitiator(frontendClient frontend.Client, numSWFn dynamicconfig.IntPropertyFn) Initiator {
+	return &initiator{
+		cadenceClient: client.NewClient(frontendClient, Domain, &client.Options{}),
+		numSWFn:       numSWFn,
+	}
+}
+
+// Archive starts an archival task
+func (i *initiator) Archive(request *ArchiveRequest) error {
+	workflowID := fmt.Sprintf("%v-%v", WorkflowIDPrefix, rand.Intn(i.numSWFn()))
```

Comment:
I think this pattern is okay. The basic idea is 

(1) We might have more system workflows then last time we read numSWFn. In this case we just start spreading new load evenly among the added system workflows
(2) If we reduced the number of system workflows, then the system workflows now out of range will sit ideal not doing anything until they expire.

Reply:
Updated from my last diff is that these workflowIDs are now stable, so the same workflows will be used upon role restart.